cognite_toolkit/cdf_tk/templates/_templates.py

```diff
@@ -377,74 +416,107 @@ def sort_key(p: Path) -> int:
+                        yaml_source_path=filepath,
+                        yaml_dest_path=destination,
+                        module_dir=module_dir,
+                        build_dir=build_dir,
+                        common_code_dir=Path(project_config_dir / environment.common_function_code),
+                        verbose=verbose,
                     )
-                    printed_function_warning = True
-
-                process_function_directory(
-                    yaml_source_path=filepath,
-                    yaml_dest_path=destination,
-                    module_dir=module_dir,
-                    build_dir=build_dir,
-                    common_code_dir=Path(project_config_dir / environment.common_function_code),
-                )
+                    all_files[resource_folder]["other_files"] = []
+                if resource_folder == "files":
+                    process_files_directory(
+                        files=all_files[resource_folder]["other_files"],
+                        yaml_dest_path=destination,
+                        module_dir=module_dir,
+                        build_dir=build_dir,
+                        verbose=verbose,
+                    )
+                    all_files[resource_folder]["other_files"] = []
+
+            if resource_folder == "timeseries_datapoints":
+                # Process all csv files
+                for filepath in all_files["timeseries_datapoints"]["other_files"]:
+                    if filepath.suffix.lower() == ".csv":
+                        # Special case for timeseries datapoints, we want to timeshift datapoints
+                        # if the file is a csv file and we have been instructed to.
+                        # The replacement is used to ensure that we read exactly the same file on Windows and Linux
+                        file_content = filepath.read_bytes().replace(b"\r\n", b"\n").decode("utf-8")
+                        data = pd.read_csv(io.StringIO(file_content), parse_dates=True, index_col=0)
+                        destination = build_dir / resource_folder / filename
+                        destination.parent.mkdir(parents=True, exist_ok=True)
+                        if "timeshift_" in data.index.name:
+                            print(
+                                "      [bold green]INFO:[/] Found 'timeshift_' in index name, timeshifting datapoints up to today..."
+                            )
+                            data.index.name = data.index.name.replace("timeshift_", "")
+                            data.index = pd.DatetimeIndex(data.index)
+                            periods = datetime.datetime.today() - data.index[-1]
+                            data.index = pd.DatetimeIndex.shift(data.index, periods=periods.days, freq="D")
+                            destination.write_text(data.to_csv())
+                        else:
+                            destination.write_text(content)
```

Comment:
`content` does not exist in this context, I think this should be
```suggestion
                        destination.write_text(data.to_csv())
```

As you are doing some modifications to the data if there is `timeshift_` in the index name, but if there is not you do the same operation.